<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />



<meta name="progressive" content="false" />
<meta name="allow-skip" content="false" />

<title>Bioconductor</title>


<!-- highlightjs -->
<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>

<!-- taken from https://github.com/rstudio/rmarkdown/blob/67b7f5fc779e4cfdfd0f021d3d7745b6b6e17149/inst/rmd/h/default.html#L296-L362 -->
<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("section-TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>
<!-- end tabsets -->



</head>

<body>



<div class="pageContent band">
<div class="bandContent page">

<div class="topics">

<div id="section-install" class="section level3">
<h3>Install</h3>
<p>You can install the <code>Bioconductor</code> package straight from CRAN. Try running the code below in your own RStudio environment. If you are asked whether to install all, some, or no dependencies, type “a” for all.</p>
<p>If you are asked if you want to update packages, say “Yes”. If a package that R tried to update can’t be updated, take note of that package’s name and install it separately from CRAN without downloading and compiling the more recent version.</p>
<p>This might take a few minutes to run.</p>
<pre class="r"><code>if (!requireNamespace(&quot;BiocManager&quot;, quietly = TRUE))
    install.packages(&quot;BiocManager&quot;)
BiocManager::install()</code></pre>
<pre><code>## Bioconductor version 3.9 (BiocManager 1.30.4), R 3.6.1 (2019-07-05)</code></pre>
<pre><code>## Update old packages: &#39;AnnotationHub&#39;, &#39;DelayedMatrixStats&#39;, &#39;devtools&#39;,
##   &#39;GenomicRanges&#39;, &#39;gsl&#39;, &#39;insight&#39;, &#39;matrixStats&#39;, &#39;plm&#39;, &#39;rtracklayer&#39;,
##   &#39;seqinr&#39;, &#39;sjstats&#39;</code></pre>
<pre class="r"><code>BiocManager::install(c(&quot;GenomicFeatures&quot;, &quot;AnnotationDbi&quot;))</code></pre>
<pre><code>## Bioconductor version 3.9 (BiocManager 1.30.4), R 3.6.1 (2019-07-05)</code></pre>
<pre><code>## Installing package(s) &#39;GenomicFeatures&#39;, &#39;AnnotationDbi&#39;</code></pre>
<pre><code>## 
## The downloaded binary packages are in
##  /var/folders/zd/3fk02lt141510y5lrtz20ldsxmdt80/T//RtmpKJV7RI/downloaded_packages</code></pre>
<pre><code>## Update old packages: &#39;AnnotationHub&#39;, &#39;DelayedMatrixStats&#39;, &#39;devtools&#39;,
##   &#39;GenomicRanges&#39;, &#39;gsl&#39;, &#39;insight&#39;, &#39;matrixStats&#39;, &#39;plm&#39;, &#39;rtracklayer&#39;,
##   &#39;seqinr&#39;, &#39;sjstats&#39;</code></pre>
<pre class="r"><code>library(BiocManager)
library()</code></pre>
<p>Now let’s see how many packages are available in <code>BiocManager</code>.</p>
<pre class="r"><code>length(BiocManager::available())</code></pre>
<pre><code>## [1] 17990</code></pre>
</div>
<div id="section-get-some-data" class="section level2">
<h2>Get some Data</h2>
<p>Install more packages. This code will work only if you have the latest R version installed.</p>
<pre class="r"><code>if (!requireNamespace(&quot;SingleCellExperiment&quot;, quietly = TRUE)) {
    BiocManager::install(&quot;SingleCellExperiment&quot;)
}
library(SingleCellExperiment)</code></pre>
<pre><code>## Loading required package: SummarizedExperiment</code></pre>
<pre><code>## Loading required package: GenomicRanges</code></pre>
<pre><code>## Loading required package: stats4</code></pre>
<pre><code>## Loading required package: BiocGenerics</code></pre>
<pre><code>## Loading required package: parallel</code></pre>
<pre><code>## 
## Attaching package: &#39;BiocGenerics&#39;</code></pre>
<pre><code>## The following objects are masked from &#39;package:parallel&#39;:
## 
##     clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,
##     clusterExport, clusterMap, parApply, parCapply, parLapply,
##     parLapplyLB, parRapply, parSapply, parSapplyLB</code></pre>
<pre><code>## The following objects are masked from &#39;package:stats&#39;:
## 
##     IQR, mad, sd, var, xtabs</code></pre>
<pre><code>## The following objects are masked from &#39;package:base&#39;:
## 
##     anyDuplicated, append, as.data.frame, basename, cbind,
##     colnames, dirname, do.call, duplicated, eval, evalq, Filter,
##     Find, get, grep, grepl, intersect, is.unsorted, lapply, Map,
##     mapply, match, mget, order, paste, pmax, pmax.int, pmin,
##     pmin.int, Position, rank, rbind, Reduce, rownames, sapply,
##     setdiff, sort, table, tapply, union, unique, unsplit, which,
##     which.max, which.min</code></pre>
<pre><code>## Loading required package: S4Vectors</code></pre>
<pre><code>## 
## Attaching package: &#39;S4Vectors&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:base&#39;:
## 
##     expand.grid</code></pre>
<pre><code>## Loading required package: IRanges</code></pre>
<pre><code>## Loading required package: GenomeInfoDb</code></pre>
<pre><code>## Loading required package: Biobase</code></pre>
<pre><code>## Welcome to Bioconductor
## 
##     Vignettes contain introductory material; view with
##     &#39;browseVignettes()&#39;. To cite Bioconductor, see
##     &#39;citation(&quot;Biobase&quot;)&#39;, and for packages &#39;citation(&quot;pkgname&quot;)&#39;.</code></pre>
<pre><code>## Loading required package: DelayedArray</code></pre>
<pre><code>## Loading required package: matrixStats</code></pre>
<pre><code>## 
## Attaching package: &#39;matrixStats&#39;</code></pre>
<pre><code>## The following objects are masked from &#39;package:Biobase&#39;:
## 
##     anyMissing, rowMedians</code></pre>
<pre><code>## Loading required package: BiocParallel</code></pre>
<pre><code>## 
## Attaching package: &#39;DelayedArray&#39;</code></pre>
<pre><code>## The following objects are masked from &#39;package:matrixStats&#39;:
## 
##     colMaxs, colMins, colRanges, rowMaxs, rowMins, rowRanges</code></pre>
<pre><code>## The following objects are masked from &#39;package:base&#39;:
## 
##     aperm, apply, rowsum</code></pre>
<pre class="r"><code>if (!requireNamespace(&quot;scRNAseq&quot;, quietly = TRUE)) {
    BiocManager::install(&quot;scRNAseq&quot;)
}
library(scRNAseq)

if (!requireNamespace(&quot;SummarizedExperiment&quot;, quietly = TRUE)) {
    BiocManager::install(&quot;SummarizedExperiment&quot;)
}
library(SummarizedExperiment)

if (!requireNamespace(&quot;ExperimentHub&quot;, quietly = TRUE)) {
    BiocManager::install(&quot;ExperimentHub&quot;)
}
library(SummarizedExperiment)

if (!requireNamespace(&quot;doParallel&quot;, quietly = TRUE)) {
    install.packages(&quot;doParallel&quot;)
}
library(doParallel)</code></pre>
<pre><code>## Loading required package: foreach</code></pre>
<pre><code>## Loading required package: iterators</code></pre>
<pre class="r"><code>if (!requireNamespace(&quot;gam&quot;, quietly = TRUE)) {
    install.packages(&quot;gam&quot;)
}
library(gam)</code></pre>
<pre><code>## Loading required package: splines</code></pre>
<pre><code>## Loaded gam 1.16.1</code></pre>
<pre class="r"><code>if (!requireNamespace(&quot;clusterExperiment&quot;, quietly = TRUE)) {
      BiocManager::install(&quot;clusterExperiment&quot;)
}</code></pre>
<pre><code>## Warning: namespace &#39;bigmemory&#39; is not available and has been replaced
## by .GlobalEnv when processing object &#39;&#39;

## Warning: namespace &#39;bigmemory&#39; is not available and has been replaced
## by .GlobalEnv when processing object &#39;&#39;</code></pre>
<pre class="r"><code>if (!requireNamespace(&quot;scone&quot;, quietly = TRUE)) {
      BiocManager::install(&quot;scone&quot;)
}</code></pre>
<pre><code>## Registered S3 method overwritten by &#39;R.oo&#39;:
##   method        from       
##   throw.default R.methodsS3</code></pre>
<pre class="r"><code>if (!requireNamespace(&quot;zinbwave&quot;, quietly = TRUE)) {
      BiocManager::install(&quot;zinbwave&quot;)
}

if (!requireNamespace(&quot;slingshot&quot;, quietly = TRUE)) {
      BiocManager::install(&quot;slingshot&quot;)
}</code></pre>
<pre><code>## Warning: namespace &#39;bigmemory&#39; is not available and has been replaced
## by .GlobalEnv when processing object &#39;&#39;

## Warning: namespace &#39;bigmemory&#39; is not available and has been replaced
## by .GlobalEnv when processing object &#39;&#39;</code></pre>
<p>Obtain the data.</p>
<pre class="r"><code># Bioconductor
library(BiocParallel)
library(clusterExperiment)
library(scone)
library(zinbwave)</code></pre>
<pre><code>## 
## Attaching package: &#39;zinbwave&#39;</code></pre>
<pre><code>## The following objects are masked from &#39;package:clusterExperiment&#39;:
## 
##     nFeatures, nSamples</code></pre>
<pre class="r"><code># GitHub
library(slingshot)</code></pre>
<pre><code>## Loading required package: princurve</code></pre>
<pre class="r"><code># CRAN
library(doParallel)
library(gam)
library(RColorBrewer)

set.seed(20)</code></pre>
<p>In the next code chunk, R checks to see if you have a data directory, and if you don’t, it creates it and stores its location in <code>data_dir</code>. Then it saves important URLs in <code>urls</code>.</p>
<pre class="r"><code>data_dir &lt;- &quot;/Users/braunsb/Documents/Arcus Learning Plan Generator/Lessons/data/&quot; 
if (!dir.exists(data_dir)) system(sprintf(&#39;mkdir %s&#39;, data_dir))

urls = c(&quot;https://www.ncbi.nlm.nih.gov/geo/download/?acc=GSE95601&amp;format=file&amp;file=GSE95601%5FoeHBCdiff%5FCufflinks%5FeSet%2ERda%2Egz&quot;,
         &quot;https://raw.githubusercontent.com/rufletch/p63-HBC-diff/master/ref/oeHBCdiff_clusterLabels.txt&quot;)</code></pre>
<p>The next code chunk contains some very good coding practice. Before downloading data from a public repository, the code checks first to see if you already have it. If not, it downloads it.</p>
<pre class="r"><code>if(!file.exists(paste0(data_dir, &quot;GSE95601_oeHBCdiff_Cufflinks_eSet.Rda&quot;))) {
  if (!dir.exists(data_dir)) system(sprintf(&#39;mkdir %s&#39;, data_dir))
  download.file(urls[1], paste0(data_dir, &quot;GSE95601_oeHBCdiff_Cufflinks_eSet.Rda.gz&quot;))
  R.utils::gunzip(paste0(data_dir, &quot;GSE95601_oeHBCdiff_Cufflinks_eSet.Rda.gz&quot;))
  # assayData(Cufflinks_eSet)$exprs = NULL
  assayData(Cufflinks_eSet)$fpkm_table = NULL
  assayData(Cufflinks_eSet)$tpm_table = NULL
  save(Cufflinks_eSet, file=&#39;data/GSE95601_oeHBCdiff_Cufflinks_eSet_reduced.Rda&#39;)
}

if(!file.exists(paste0(data_dir, &quot;oeHBCdiff_clusterLabels.txt&quot;))) {
  download.file(urls[2], paste0(data_dir, &quot;oeHBCdiff_clusterLabels.txt&quot;))
}</code></pre>
<p>We load the data into <code>Cufflinks_eSet</code>, establish a count matrix, then remove undetected genes from the data.</p>
<pre class="r"><code>load(paste0(&quot;/Users/braunsb/Documents/Arcus Learning Plan Generator/data/GSE95601_oeHBCdiff_Cufflinks_eSet.Rda&quot;))

# Count matrix
E &lt;- assayData(Cufflinks_eSet)$counts_table

# Remove undetected genes
E &lt;- na.omit(E)
E &lt;- E[rowSums(E)&gt;0,]
dim(E)</code></pre>
<pre><code>## [1] 28361   849</code></pre>
<p>At this point, I think we’ve gotten you past any code in which what you do and what the website instructions say to do are different. You can continue by going to <a href="https://www.bioconductor.org/help/course-materials/2017/BioC2017/Day2/Workshops/singleCell/doc/workshop.html#pre-processing">this page on the bioconductor website</a> and searching for “We remove the ERCC spike-in sequences and the CreER gene”. Use the instructions there to do this workflow.</p>
<p>Please enter your new code into code chunks below this point, and let me know if you have any questions.</p>
<!--html_preserve-->
<script type="application/shiny-prerendered" data-context="dependencies">
{"type":"list","attributes":{},"value":[{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["jquery"]},{"type":"character","attributes":{},"value":["1.11.3"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["rmd/h/jquery"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["jquery.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["rmarkdown"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["1.15"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["bootstrap"]},{"type":"character","attributes":{},"value":["3.3.5"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["rmd/h/bootstrap"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["viewport"]}},"value":[{"type":"character","attributes":{},"value":["width=device-width, initial-scale=1"]}]},{"type":"character","attributes":{},"value":["js/bootstrap.min.js","shim/html5shiv.min.js","shim/respond.min.js"]},{"type":"character","attributes":{},"value":["css/cerulean.min.css"]},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["rmarkdown"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["1.15"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["pagedtable"]},{"type":"character","attributes":{},"value":["1.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["rmd/h/pagedtable-1.1"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["js/pagedtable.js"]},{"type":"character","attributes":{},"value":["css/pagedtable.css"]},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["rmarkdown"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["1.15"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["highlightjs"]},{"type":"character","attributes":{},"value":["9.12.0"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["rmd/h/highlightjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["highlight.js"]},{"type":"character","attributes":{},"value":["textmate.css"]},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["rmarkdown"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["1.15"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["tutorial"]},{"type":"character","attributes":{},"value":["0.9.2.9006"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/tutorial"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["tutorial.js"]},{"type":"character","attributes":{},"value":["tutorial.css"]},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.9.2.9006"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["tutorial-autocompletion"]},{"type":"character","attributes":{},"value":["0.9.2.9006"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/tutorial"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["tutorial-autocompletion.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.9.2.9006"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["tutorial-diagnostics"]},{"type":"character","attributes":{},"value":["0.9.2.9006"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/tutorial"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["tutorial-diagnostics.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.9.2.9006"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["tutorial-format"]},{"type":"character","attributes":{},"value":["0.9.2.9006"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["rmarkdown/templates/tutorial/resources"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["tutorial-format.js"]},{"type":"character","attributes":{},"value":["tutorial-format.css","rstudio-theme.css"]},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.9.2.9006"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["jquery"]},{"type":"character","attributes":{},"value":["1.11.3"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["rmd/h/jquery"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["jquery.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["rmarkdown"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["1.15"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["navigation"]},{"type":"character","attributes":{},"value":["1.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["rmd/h/navigation-1.1"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["tabsets.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["rmarkdown"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["1.15"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["highlightjs"]},{"type":"character","attributes":{},"value":["9.12.0"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["rmd/h/highlightjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["highlight.js"]},{"type":"character","attributes":{},"value":["default.css"]},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["rmarkdown"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["1.15"]}]}]}
</script>
<!--/html_preserve-->
<!--html_preserve-->
<script type="application/shiny-prerendered" data-context="execution_dependencies">
{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["packages"]}},"value":[{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["packages","version"]},"class":{"type":"character","attributes":{},"value":["data.frame"]},"row.names":{"type":"integer","attributes":{},"value":[1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39,40,41,42,43,44,45,46,47,48,49,50,51,52,53,54,55,56,57,58,59,60,61,62,63,64,65,66,67,68,69,70,71,72,73,74,75,76,77,78,79,80,81,82,83,84,85,86,87,88,89,90,91,92,93,94,95,96,97,98,99,100,101,102,103,104,105,106,107,108,109,110,111,112,113,114,115,116,117,118,119,120,121,122,123,124,125,126,127,128,129,130,131,132,133,134,135,136,137,138,139,140,141,142,143,144,145,146,147,148,149,150,151,152,153,154,155,156,157,158,159,160,161,162,163,164,165,166,167,168,169,170,171,172,173,174,175,176,177,178,179,180,181,182,183,184,185,186,187,188,189,190,191,192,193,194,195]}},"value":[{"type":"character","attributes":{},"value":["ade4","ADGofTest","annotate","AnnotationDbi","AnnotationHub","ape","aroma.light","assertthat","backports","base","bayesm","bibtex","Biobase","BiocFileCache","BiocGenerics","BiocManager","BiocParallel","biomaRt","Biostrings","bit","bit64","bitops","blob","boot","caTools","class","cluster","clusterExperiment","codetools","colorspace","compiler","compositions","copula","crayon","crosstalk","curl","datasets","DBI","dbplyr","DelayedArray","DEoptimR","DESeq","digest","diptest","doParallel","dplyr","EDASeq","edgeR","energy","evaluate","ExperimentHub","flexmix","foreach","fpc","gam","gdata","genefilter","geneplotter","GenomeInfoDb","GenomeInfoDbData","GenomicAlignments","GenomicFeatures","GenomicRanges","ggplot2","glmnet","glue","gplots","graphics","grDevices","grid","gridBase","gsl","gtable","gtools","HDF5Array","hexbin","hms","howmany","htmltools","htmlwidgets","httpuv","httr","hwriter","igraph","interactiveDisplayBase","IRanges","iterators","jsonlite","kernlab","KernSmooth","knitr","later","lattice","latticeExtra","lazyeval","learnr","limma","locfdr","locfit","magrittr","manipulateWidget","markdown","MASS","Matrix","matrixStats","mclust","memoise","methods","mime","miniUI","mixtools","modeltools","munsell","mvtnorm","nlme","NMF","nnet","numDeriv","parallel","pcaPP","phylobase","pillar","pkgconfig","pkgmaker","plyr","prabclus","prettyunits","princurve","progress","promises","pspline","purrr","R.methodsS3","R.oo","R.utils","R6","rappdirs","rARPACK","RColorBrewer","Rcpp","RCurl","registry","reshape2","rgl","rhdf5","Rhdf5lib","rlang","rmarkdown","rncl","RNeXML","rngtools","robustbase","rprojroot","Rsamtools","RSpectra","RSQLite","rtracklayer","RUVSeq","S4Vectors","scales","scone","scRNAseq","segmented","shiny","ShortRead","SingleCellExperiment","slingshot","softImpute","splines","stabledist","stats","stats4","stringi","stringr","SummarizedExperiment","survival","tensorA","tibble","tidyr","tidyselect","tools","utils","uuid","vctrs","webshot","withr","xfun","XML","xml2","xtable","XVector","yaml","zeallot","zinbwave","zlibbioc"]},{"type":"character","attributes":{},"value":["1.7-13","0.3","1.62.0","1.46.1","2.16.0","5.3","3.14.0","0.2.1","1.1.4","3.6.1","3.1-3","0.4.2","2.44.0","1.8.0","0.30.0","1.30.4","1.18.1","2.40.4","2.52.0","1.1-14","0.9-7","1.0-6","1.2.0","1.3-23","1.17.1.2","7.3-15","2.1.0","2.4.4","0.2-16","1.4-1","3.6.1","1.40-2","0.999-19.1","1.3.4","1.0.0","4.0","3.6.1","1.0.0","1.4.2","0.10.0","1.0-8","1.36.0","0.6.20","0.75-7","1.0.15","0.8.3","2.18.0","3.26.8","1.7-6","0.14","1.10.0","2.3-15","1.4.7","2.2-3","1.16.1","2.18.0","1.66.0","1.62.0","1.20.0","1.2.1","1.20.1","1.36.4","1.36.0","3.2.1","2.0-18","1.3.1","3.0.1.1","3.6.1","3.6.1","3.6.1","0.4-7","1.9-10.3","0.3.0","3.8.1","1.12.2","1.27.3","0.5.1","0.3-1","0.3.6","1.3","1.5.1","1.4.1","1.3.2","1.2.4.1","1.22.0","2.18.2","1.0.12","1.6","0.9-27","2.23-15","1.24","0.8.0","0.20-38","0.6-28","0.2.2","0.9.2.9006","3.40.6","1.1-8","1.5-9.1","1.5","0.10.0","1.1","7.3-51.4","1.2-17","0.54.0","5.4.5","1.1.0","3.6.1","0.7","0.1.1.1","1.1.0","0.2-22","0.5.0","1.0-11","3.1-141","0.21.0","7.3-12","2016.8-1.1","3.6.1","1.9-73","0.8.6","1.4.2","2.0.2","0.27","1.8.4","2.3-1","1.0.2","2.1.4","1.2.2","1.0.1","1.0-18","0.3.2","1.7.1","1.22.0","2.9.0","2.4.0","0.3.1","0.11-0","1.1-2","1.0.2","1.95-4.12","0.5-1","1.4.3","0.100.30","2.28.0","1.6.0","0.4.0","1.15","0.8.3","2.3.0","1.4","0.93-5","1.3-2","2.0.0","0.15-0","2.1.2","1.44.3","1.18.0","0.22.0","1.0.0","1.8.0","1.10.0","1.0-0","1.3.2","1.42.0","1.6.0","1.2.0","1.4","3.6.1","0.7-1","3.6.1","3.6.1","1.4.3","1.4.0","1.14.1","2.44-1.1","0.36.1","2.1.3","0.8.3","0.2.5","3.6.1","3.6.1","0.1-2","0.2.0","0.5.1","2.1.2","0.9","3.98-1.20","1.2.2","1.8-4","0.24.0","2.2.0","0.1.0","1.6.0","1.30.0"]}]}]}
</script>
<!--/html_preserve-->
</div>

</div> <!-- topics -->

<div class="topicsContainer">
<div class="topicsPositioner">
<div class="band">
<div class="bandContent topicsListContainer">

<!-- begin doc-metadata -->
<div id="doc-metadata">
<h2 class="title toc-ignore" style="display:none;">Bioconductor</h2>
</div>
<!-- end doc-metadata -->

</div> <!-- bandContent.topicsListContainer -->
</div> <!-- band -->
</div> <!-- topicsPositioner -->
</div> <!-- topicsContainer -->


</div> <!-- bandContent page -->
</div> <!-- pageContent band -->




<script>
// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});
</script>


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>


</body>

</html>
